command/agent/agent_auto_auth_self_heal_test.go

```diff
@@ -447,3 +424,22 @@ func waitForFiles(t *testing.T, filePath string, prevModTime time.Time) (os.File
 		return fileInfo, nil
 	}
 }
+
+// makeTempFile creates a temp file with the specified name, populates it with the
+// supplied contents and closes it. The path to the file is returned, also the file
+// will be automatically removed when the test which created it, finishes.
+func makeTempFile(t *testing.T, name, contents string) string {
```

Comment:
I really, really wanted to refactor `makeTempFile` to be in our testhelpers, since we use it here and in `agent.go` and `proxy.go`. In fact, that was much of the time I spent for this PR. However, we use `github.com/mitchellh/go-testing-interface` there to make these helpers work, and that library is both deprecated and doesn't support `t.TempDir()` which is the main benefit of this helper. It would have been a big lift to update all of our helpers, and I felt like it was out of scope somewhat.

 That being said, if we ever do factor this out, it'll be easy to replace :)